posthog/hogql/hogql.py

```diff
@@ -19,8 +19,14 @@ def translate_hogql(query: str, context: HogQLContext, dialect: Literal["hogql",
         context.database = context.database or create_hogql_database(context.team_id)
         select_query_ref = ast.SelectQueryRef(tables={"events": ast.TableRef(table=context.database.events)})
         node = parse_expr(query, no_placeholders=True)
-        select_query = ast.SelectQuery(select=[node], ref=select_query_ref)
-        return print_ast(node, context=context, dialect=dialect, stack=[select_query])
+        select_query = ast.SelectQuery(
+            select=[node], select_from=ast.JoinExpr(table=ast.Field(chain=["events"])), ref=select_query_ref
+        )
+        prepared_ast: ast.SelectQuery = cast(
+            ast.SelectQuery,
+            prepare_ast_for_printing(select_query, context=context, dialect=dialect, stack=[select_query]),
+        )
+        return print_prepared_ast(prepared_ast.select[0], context=context, dialect=dialect, stack=[select_query])
```

Comment:
This `translate_hogql` function is only used to print HogQL filters inside legacy insights. There was a "undefined ref" bug that these lines fix. Basically we create a virtual "select" query, and print its first field. If the column itself got "swapped" (e.g. by adding `toInt()` around it), then the mutated object wouldn't get transferred into the previous virtual select. The changes above fix that --> now we prepare the entire query manually here, and then only print out the actual first select column. 

Hope that makes sense 😅